gitindex/index.go

```diff
@@ -522,7 +501,27 @@ func indexGitRepo(opts Options, config gitIndexConfig) error {
 		names = append(names, n)
 	}
 
-	sort.Strings(names)
+	if opts.BuildOptions.OfflineRankingEnabled() {
+		data, err := os.ReadFile(opts.BuildOptions.DocumentRanksPath)
+		if err != nil {
+			return err
+		}
+
+		var ranks map[string][]float64
+		err = json.Unmarshal(data, &ranks)
+		if err != nil {
+			return err
+		}
+
+		// We sort documents according to their rank before we start adding them to the
+		// builder. This means that the first shard will contain the documents with the
+		// highest rank.
+		sortDocuments2(names, ranks)
+
+	} else {
+		sort.Strings(names)
+	}
+
 	names = uniq(names)
```

Comment:
won't this fail now since it isn't sorted lexicographically? Alternativelly it likely will work as long as the same elements are next to each other, but then you need to make sure in sortDocuments2 we do a final tie break by name.

Reply:
It will fail if there are 2 documents with exactly the same rank, which is possible, especially for smaller repos.